### PR TITLE
[RC1] Admin Bid Cycles

### PR DESCRIPTION
**Pull request by Icemanbeta**
- Created `Base` component for extending capabilities and reducing code base with reusable functionality
- Created `Card` component wrapper that renders with **card** class for basic uses and reuse
- Add mock for `/api/v1/bidCycles/`
- Created container -> component tree for `BidCycles`

### CODE REVIEW

File: src/Components/BidCycles/BidCycle/BidCycle.jsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+const BidCycle = (props) => {
+  const cycle = {
+    name: getCycleInfo(props.cycle, 'name'),
+    start: getCycleInfo(props.cycle, 'cycle_start_date'),
+    end: getCycleInfo(props.cycle, 'cycle_end_date'),
+    deadline: getCycleInfo(props.cycle, 'cycle_deadline_date'),
+  };
+
+  return (cycle ?
+    <Card>
```
Comment (mjoyce91) [inline src/Components/BidCycles/BidCycle/BidCycle.jsx]: Can we get these to not overlap?

screen shot 2018-03-29 at 1 05 14 pm

File: src/Containers/BidCycles/BidCycles.jsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import { bindActionCreators } from 'redux';
+import { pick } from 'lodash';
+import PropTypes from 'prop-types';
+
+import { fetchBidCycles } from '../../actions/bidCycles';
+import BidCycleComponent from '../../Components/BidCycles';
+import { BID_CYCLES } from '../../Constants/PropTypes';
+
+class BidCycles extends Component {
+  static propTypes = {
```
Comment (mjoyce91) [inline src/Containers/BidCycles/BidCycles.jsx]: For some reason setting `propTypes` with `static` is giving me compile errors:

```
./src/Containers/BidCycles/BidCycles.jsx
  Line 12:  'propTypes' is not defined     no-undef
  Line 19:  'defaultProps' is not defined  no-undef

Search for the keywords to learn more about each error.
```

Is there any benefit to setting them this way, or could we set them outside of the class?

File: src/Containers/BidCycles/BidCycles.test.jsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  <Provider store={mockStore({ bidCycles: [] })}>
+    <MemoryRouter>
+      <BidCycles />
+    </MemoryRouter>
+  </Provider>
+);
+
+describe('BidCyclesContainer', () => {
+  beforeEach(() => {
+    mockAdapter.onAny().reply((config) => {
```
Comment (mjoyce91) [inline src/Containers/BidCycles/BidCycles.test.jsx]: Didn't know about `onAny`! :+1:

File: src/Components/BidCycles/BidCycle/BidCycle.jsx (hunk 1 above)
Comment (Icemanbeta) [reply src/Components/BidCycles/BidCycle/BidCycle.jsx]: Dag, I must've screwed up using the usa-grids

File: src/Components/BidCycles/BidCycle/BidCycle.jsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  const defaults = {
+    name: NO_BID_CYCLE,
+    cycle_start_date: NO_DATE,
+    cycle_end_date: NO_DATE,
+    cycle_deadline_date: NO_DATE,
+  };
+
+  let item$ = _.get(item, key, defaults[key]);
+
+  if (key === 'name') {
```
Comment (burgwyn) [inline src/Components/BidCycles/BidCycle/BidCycle.jsx]: We can remove this logic.  Most bid cycles are "Spring 2018" or "Training & Language"

File: src/Components/BidCycles/BidCycles.jsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+import { find } from 'lodash';
+import BidCycle from './BidCycle';
+import { Row, Column } from '../Layout';
+import ProfileSectionTitle from '../ProfileSectionTitle';
+import { BID_CYCLES } from '../../Constants/PropTypes';
+import Menu from '../../Constants/Menu';
+
+const title = find(Menu[0].children, { route: '/profile/cycles/' }).text;
+
+const BidCycles = ({ cycles }) => (
```
Comment (burgwyn) [inline src/Components/BidCycles/BidCycles.jsx]: Can we rename to `BidCycleList`

File: src/Components/BidCycles/BidCycle/BidCycle.jsx [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+  }
+
+  if ((/date/).test(key)) {
+    item$ = formatDate(item$, 'M.D.YYYY');
+  }
+
+  return item$;
+};
+
+const BidCycle = (props) => {
```
Comment (burgwyn) [inline src/Components/BidCycles/BidCycle/BidCycle.jsx]: Can we rename to something like `BidCycleCard` or `BidCycleListItem`

File: src/Components/BidCycles/BidCycle/BidCycle.jsx (hunk 4 above)
Comment (Icemanbeta) [reply src/Components/BidCycles/BidCycle/BidCycle.jsx]: Will do, this was for the data coming in from the API, I totally understand if it won't be like this:
image

**Comment by mjoyce91** [comment]
@burgwyn @Icemanbeta - it looks like the link to this is visible to all users. We should limit it to only an admin.

**Pull request merged**